### PR TITLE
Sample project reviving

### PR DESCRIPTION
**Pull request by Javernaut**
The Sample application is back.

Main features:

- Fully converted to Kotlin
- Is in sync with the latest Cicerone API (6.1)
- Moxy dependency was changed to the one that supports AndroidX
- Other dependencies and build tools are updated
- The applicationId was migrated to 'com.github.terrakok.cicerone.sample'

### CODE REVIEW

**Comment by Javernaut** [comment]
Please sync the develop branch with the master. Otherwise this PR is bloated with irrelevant changes.

File: sample/build.gradle [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+dependencies {
+    // Support libraries
+    implementation "androidx.appcompat:appcompat:$androidXVersion"
+    implementation "com.google.android.material:material:$materialVersion"
+
+    //MVP Moxy
+    implementation "tech.schoolhelper:moxy-x:$moxyVersion"
+    implementation "tech.schoolhelper:moxy-x-androidx:$moxyVersion"
+    kapt "tech.schoolhelper:moxy-x-compiler:$moxyVersion"
```
Comment (terrakok) [inline sample/build.gradle:50]: you should use https://github.com/moxy-community/Moxy

File: sample/build.gradle [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    //DI
+    implementation "com.google.dagger:dagger:$daggerVersion"
+    kapt "com.google.dagger:dagger-compiler:$daggerVersion"
+
+    //Bottom Navigation Bar
+    implementation ('com.ashokvarma.android:bottom-navigation-bar:1.3.0') {
+        exclude group: "com.android.support", module: "design"
+    }
+    implementation "androidx.core:core-ktx:1.3.2"
+    implementation "org.jetbrains.kotlin:kotlin-stdlib-jdk7:$kotlin_version"
```
Comment (terrakok) [inline sample/build.gradle:64]: kotlin plugin adds dependency on stdlib automaticaly

**Review by terrakok — CHANGES_REQUESTED** [review]

File: sample/build.gradle (hunk 1 above)
Comment (Javernaut) [reply sample/build.gradle]: I've done the migration

File: sample/build.gradle (hunk 2 above)
Comment (Javernaut) [reply sample/build.gradle]: That one was added by Android Studio. I removed that line.

**Comment by Javernaut** [comment]
I've done the requested changes. Please, review again.

**Pull request merged**